### ISSUE TITLE
Using DSet<bool> cause exception at Prajna.Core.MetaFunction`1.EncodeFunc

### ISSUE THREAD

**Issue by zhuweirong** [MSRCCS/Prajna#106]
For code like

```
    let numPartitions = 4
    let guid = Guid.NewGuid().ToString("D")
    let d = DSet<_> ( Name = guid, Cluster = cluster)
           |> DSet.sourceI numPartitions (fun i -> seq { for i = 0 to 9 do 
                                                            if i % 2 = 0 then yield true else yield false})

    let r = d.ToSeq() |> Array.ofSeq
```

When run it with local cluster, an exception was thrown in serializer. 

The exception is

```
  System.ArgumentException: Offset and length were out of bounds for the array or count is greater than the number of elements from index to the end of the source collection
```

The stack is

```
   at System.Buffer.BlockCopy(Array src, Int32 srcOffset, Array dst, Int32 dstOffset, Int32 count)
   at Prajna.Tools.BufferListStream`1.SrcDstBlkCopy[T1,T2,T1Elem,T2Elem](T1 src, Int32& srcOffset, Int32& srcLen, T2 dst, Int32& dstOffset, Int32& dstLen) in C:\GitHub\Prajna\src\tools\tools\bufferliststream.fs:line 994
   at Prajna.Tools.BufferListStream`1.WriteArrT(Array buf, Int32 offset, Int32 count) in C:\GitHub\Prajna\src\tools\tools\bufferliststream.fs:line 1353
   at Prajna.Tools.Serializer.writeMemoryBlittableArray(Type elType, Array arrObj, MemoryStream memStream) in C:\GitHub\Prajna\src\tools\tools\serialize.fs:line 348
   at <StartupCode$Prajna-Tools>.$Serialize.-ctor@379-28.Invoke(Tuple`4 tupledArg) in C:\GitHub\Prajna\src\tools\tools\serialize.fs:line 379
   at Prajna.Tools.Serializer.WriteArray(Type arrayType, Array arrObj) in C:\GitHub\Prajna\src\tools\tools\serialize.fs:line 410
   at Prajna.Tools.BinarySerializer.System-Runtime-Serialization-IFormatter-Serialize(Stream stream, Object graph) in C:\GitHub\Prajna\src\tools\tools\serialize.fs:line 745
   at Prajna.Core.MetaFunction`1.EncodeFunc(BlobMetadata meta, U[] elemArray) in C:\GitHub\Prajna\src\CoreLib\function.fs:line 275
   at Prajna.Core.MetaFunction`1.EncodeFuncFromObj(BlobMetadata meta, Object o) in C:\GitHub\Prajna\src\CoreLib\function.fs:line 253
```

**Comment by jinlmsft** [comment]
Sanjeev,

Can you take a quick look or chat with Weirong? Is the issue again the need to call Environment.Init() beforehand?

Best,
Jin

**Comment by zhuweirong** [comment]
Resolved with #108
